### ISSUE TITLE
Helm Charts for SubQuery Network Services

### ISSUE THREAD

**Issue by jamesbayly** [subquery/network-indexer-services#422]
## Context:

Currently we only document how to run our Node Operator Services (`https://raw.githubusercontent.com/subquery/network-indexer-services`) on [single node vm](https://academy.subquery.network/subquery_network/node_operators/setup/install-linux.html) or using [Docker](https://academy.subquery.network/subquery_network/node_operators/setup/install-local-docker.html), while some node operators wanted to run it on kubernetes.

We should be able to provide a limited featured version and make it work with kubernetes. This limited feature edition will only support Node Operators with RPC projects.

## Requirements:

- Fix compatibility issues
  - find another way to show coordinator’s std output or disable it
  - disable the feature to create managed subql projects
- Create a helmchart to help people deploy the whole stack
  - indexer proxy
  - coordinator
  - ipfs
  - ingress

**Comment by Faizalhidayat** [comment]
thsi task only create helm chart ?

**Comment by Faizalhidayat** [comment]
I have created a Helm chart that .

try this https://github.com/Faizalhidayat/subquery-helm-chart.git

**Comment by icezohu** [comment]
Hi @Faizalhidayat ,
Thank you for your efforts in resolving this issue. We will schedule a detailed review of the Helm charts soon. As for the task, it should also fix the compatibility issues mentioned above. Will you contribute to these?

**Comment by Faizalhidayat** [comment]
Thank you for your appreciation and for considering me for this task. I’m glad to hear that a detailed review of the Helm charts is being scheduled.

**Comment by jamesbayly** [comment]
@Faizalhidayat you are on the right track, but we need you to complete the listed compatibility issues before we can start reviewing.

Without fixing these issues, these helm charts will not work correctly

**Comment by Faizalhidayat** [comment]
detail what problem I haven't solved yet??

**Comment by jamesbayly** [comment]
> detail what problem I haven't solved yet??  This task only focus create to helm chart subquery ???

Fix compatibility issues
- find another way to show coordinator’s std output or disable it
- disable the feature to create managed subql projects

**Comment by Faizalhidayat** [comment]
i already update my helm chart following issue above

**Comment by icezohu** [comment]
Hi, @Faizalhidayat 

- find another way to show coordinator’s std output or disable it
we show coordinator and index node std output through docker sock, which is displaying on the admin UI. I think k8s does not allow to expose its admin interface to a coodinator pod. I think the simple way is to hide them in admin UI.

- disable the feature to create managed subql projects
creating a subql project will throw error in k8s pod, as you may know, we used to create container through docker sock. The simple way is to reject create subql projects when input subql cid into the admin UI field.

**Comment by Faizalhidayat** [comment]
Any update ??

**Comment by Faizalhidayat** [comment]
@icezohu please?

**Comment by icezohu** [comment]
Sorry, according to my prev suggestion, the code should be more related to the frontend app code at `network-indexer-services/apps/indexer-admin`

**Comment by icezohu** [comment]
Yes, or may be including backend part as well. That may require more of the project background knowledge. Try if you can. Tell us if it's difficult for you to implement.

**Comment by Faizalhidayat** [comment]
lets point this task . 

1. Fix compatibility issues
find another way to show coordinator’s std output or disable it
disable the feature to create managed subql projects on network-indexer-services/apps/indexer-admin , right?

2.Create a helmchart to help people deploy the whole stack
indexer proxy
coordinator
ipfs
ingress

its different task , right ?

**Comment by Faizalhidayat** [comment]
@icezohu can you answer my question above

**Comment by icezohu** [comment]
@Faizalhidayat Yes, they are different. We just release a new version of coordinator yesterday, which supports adding self hosted subquery projects. I think it's better to provide a startup parameter in coordinator/typescript just to disable adding coordinator hosted subquery projects and disable coordinator logs.
Btw, is your helm charts tested well? I can't start it up straightforwardly or after some tweaking. And also the startup command you provided in readme is not correct, though this does not really matter.

**Comment by Faizalhidayat** [comment]
last question 


1.find another way to show coordinator’s std output or disable it
disable the feature to create managed subql projects
on  https://github.com/subquery/network-indexer-services , right? 

2. create helm chart Create a helmchart to help people deploy the whole stack for kubernetes ?
indexer proxy
coordinator
ipfs
ingress

**Comment by Faizalhidayat** [comment]
hi @icezohu can you check my pr 
https://github.com/subquery/network-indexer-services/pull/504

**Comment by Faizalhidayat** [comment]
@icezohu

**Comment by pamanseau** [comment]
I will do the Helm chart but I have compatibility issues #527

**Comment by pamanseau** [comment]
@Faizalhidayat 

I see `admin-ui-configmap.yaml` that creates a config.json but this is not mounted anywhere.

```
  config.json: |
    {
      "disableManagedProjects": {{ .Values.coordinator.disableManagedProjects }},
      "showCoordinatorStdout": {{ .Values.coordinator.showStdoutInAdminUI }},
      "environment": "{{ .Values.global.environment }}",
      "coordinatorUrl": "http://{{ include "my-subql-chart.fullname" . }}-coordinator:{{ .Values.coordinator.service.port }}"
    }
```
I didn't see in the code or docker-compose.yaml this file or any documentation about this.

**Comment by icezohu** [comment]
As k8s support for coordinator service is implemented in #529 , closing this issue accordingly.

**Comment by Faizalhidayat** [comment]
@icezohu how about payment?
